packages/fxa-shared/subscriptions/metadata.ts

```diff
@@ -35,20 +35,31 @@ export const DEFAULT_PRODUCT_DETAILS: ProductDetails = {
 
 // Support some default null values for product / plan metadata and
 // allow plan metadata to override product metadata
-export const metadataFromPlan = (plan: Plan): ProductMetadata => ({
-  productSet: null,
-  productOrder: null,
-  emailIconURL: null,
-  webIconURL: null,
-  webIconBackground: null,
-  upgradeCTA: null,
-  successActionButtonURL: null,
-  'product:termsOfServiceDownloadURL': '',
-  'product:termsOfServiceURL': '',
-  'product:privacyNoticeURL': '',
-  ...plan.product_metadata,
-  ...plan.plan_metadata,
-});
+export const metadataFromPlan = (plan: Plan): ProductMetadata => {
+  const metadata = {
+    productSet: null,
+    productOrder: null,
+    emailIconURL: null,
+    webIconURL: null,
+    webIconBackground: null,
+    upgradeCTA: null,
+    successActionButtonURL: null,
+    'product:termsOfServiceDownloadURL': '',
+    'product:termsOfServiceURL': '',
+    'product:privacyNoticeURL': '',
+    ...plan.product_metadata,
+    ...plan.plan_metadata,
+  };
+
+  if (!!metadata.productSet) {
```

Comment:
Q: What does `!!metadata.productSet` do/guard?
It seems to be a truthy check, sure, but not 100% certain what it's checking for; considering the following line is checking an explicit `typeof metadata.productSet === 'string'`.
Without the extra `if (!!metadata.productSet) {...}` wouldn't this still fail:
```js
if (typeof null === 'string') {...}
```

If I check it in the Node 16.17.0 REPL, I get:
```js
typeof null // 'object'
typeof undefined // 'undefined'
```

Unless this is guarding against empty strings:
```js
!!'' // false
typeof '' // 'string'
```

Reply:
removed this check